scripts/build_gh_pages

```diff
@@ -4,71 +4,70 @@ readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
 # shellcheck source=scripts/common
 source "$SCRIPTS_DIR/common"
 
-readonly SRC_SUBDIRS=(sdk/rust oak/server/rust)
-readonly DOC_SUBDIRS=(sdk server)
-
 # Update the gh-pages branch. Note that `cargo doc` is **not deterministic** so
 # this should only be done when there is a real change.
-readonly BRANCH=${1:-master}
-readonly PREV_COMMIT=$(git log --oneline -n 1 gh-pages | sed 's/.*branch at \([0-9a-f]*\)/\1/')
 
-readonly CHANGES=$(git diff "${PREV_COMMIT}..${BRANCH}" -- "${SRC_SUBDIRS[@]}" | grep -e '[+-]//[/!]')
+# First check if the current commit is "clean".
```

Comment:
This still has the problem that running on any non-`master` branch will generate a `gh-pages` commit referencing an orphaned Git hash.

Reply:
Amended now, PTAL.